lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -86,14 +87,15 @@
  * An LL(*) recursive-descent parser for the context-free grammar of the Doctrine Query Language.
  * Parses a DQL query, reports any errors in it, and generates an AST.
  *
+ * @psalm-import-type Token from AbstractLexer
```

Comment:
Should we bump `doctrine/lexer`'s version constraint?

Reply:
Just for static analysis? Feels unnecessary.

Reply:
Is it a big deal of bumping it? Also, isn't there a risk of breaking our users' SA pipeline? What will happen if they upgrade the ORM without upgrading `doctrine/lexer`?

Reply:
Note that 1.2.3 supports PHP 7.1 so I wouldn't say it's a big deal.

Reply:
> Is it a big deal of bumping it?

No, we could do it. It's just that we don't really need to.

> Also, isn't there a risk of breaking our users' SA pipeline? What will happen if they upgrade the ORM without upgrading `doctrine/lexer`?

Most projects won't notice, unless they implement their own tree walkers maybe.